docs/usage_guide/optimization_guide/index.rst

```diff
@@ -0,0 +1,58 @@
+Optimization Guide
+==================
+
+Here you can find information on Modin optimizations both for general query pipeline and for specific operations.
+
+Generic settings
+""""""""""""""""
+
+If you are working with a wide DataFrame and performing an operation that requires ``map`` Core Dataframe Algebra operator
+to be applied cell-wise, you may want to change the number of partitions DataFrame should partition into to avoid
+excessive amount of remote partitions, which is much greater than the number of CPUs.
```

Comment:
It's not clear why a user may observe "an excessive amount of remote partitions". Do we have a section in docs, describing the partitioning strategy and that it may result in NCORES^2 amount partitions? If so, I'd like to link it here. If there's no such section, I would briefly describe the partitioning mechanism itself here, on this page.